projects/week_2/Instructions.md

```diff
@@ -0,0 +1,275 @@
+
+Next, declare another `cv::Mat` variable that will hold our output image, named `output`. We'll initialize this variable to be the same size as our input image while holding one channel of unsigned, 8-bit values.
+
+```c++
+cv::Mat output(input.size(), CV_8UC1);
+```
+
+Now we need to iterate over each pixel position in our images. Create two nested loops. The first loop should iterate over each row index, from 0 to `input_hsv.rows`. The second loop should iterate over each column index, from 0 to `input_hsv.cols`. You can name the row index variable `r` and the column index variable `c`.
+
+In the body of our column loop, we can get the color at that position using the `at` function of `cv::Mat`:
+
+```c++
+const auto input_color = input_hsv.at<cv::Vec3b>(r,c);
+```
+
+Note that we're asking for the value at that position as a `cv::Vec3b` so it's easy to compare against our range variables. Each `cv::Vec3b` holds the three channel values that represent the color value at a given pixel as a vector (in the math sense, not `std::vector`) of bytes. To compare scalars, we can compare each of the channel values individually. `cv::Vec3b` provides a square bracket operator so we can access each channel value by its index. For example, to check if channel 0 of our input color is in our target range, we could write this:
+
+```c++
+input_color[0] >= range_min[0] && input_color[0] <= range_max[0]
+```
+
+Create an if statement that checks if all three channels of `input_color` are within our target range. Inside the body of this if statment, set the current position of our output image to 255:
+
+```c++
+output.at<uint8_t>(r,c) = 255;
+```
+
+Add an else block to our if statement. In this branch, set the output value to 0.
+
+Finally, after the end of the nested loops, return our `output` image.
+
+### 3.4 Implement `ReprojectToGroundPlane`
+
+`ReprojectToGroundPlane` warps `input` using the homography defined by the `homography` matrix. This function will create an output image with the size given by `map_size`. It will then iterate over each pixel position in the output image and calculate the corresponding pixel position in the input image with the homography matrix. Finally, it copies the value from the source position in the source image to the destination position in the output image. Any pixels that get mapped outside of the bounds of the source image will be set to 127 in the output image.
+
+Let's start by declaring our output image variable. We'll call it `output` and use `cv::Mat` as its type. Initialize it by passing `map_size` and `CV_8UC1` to its constructor. `map_size` just holds the desired height and width for our output image. `CV_8UC1` tells OpenCV that this image will have one channel of unsigned, 8-bit values. (Our color images so far have been using `CV_8UC3` to store three channels of unsigned, 8-bit values).
```

Comment:
bits are explained better here then when they are first mentioned.

Reply:
Moved this explanation up to the first use of `CV_8UC1`.